clusterloader2/api/validation/validation_test.go

```diff
@@ -441,3 +442,29 @@ func TestValidate(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateMeasurement(t *testing.T) {
+	for _,test := range []struct {
+		name   string
+		input  api.Measurement
+		expected *errors.ErrorList
+	}{
+		{
+			name: "first-test",
+			input: api.Measurement{
+				Method : "APIAvailability",
+				Identifier : "APIAvailability",
+			},
+			expected:nil,
+		},
+	}{
+		t.Run(test.name,func(t *testing.T) {
+			v := NewConfigValidator("", &api.Config{})
```

Comment:
@marseel , so while writing the unit tests , how do I register the measurement, 
I know in factory there is a `register` method , but it takes second argument `createFunc createMeasurementFunc` , so not sure how do I call that function?
Otherwise every time test runs, it says `measurement not registered`.

Reply:
You  need to create function that returns empty measurement and pass it to register function. This function is defined as: `type createMeasurementFunc func() Measurement`

Reply:
So , should that empty measurement also imlement the three methods, `Execute`, `String` and `Dispose` which will probably empty?

Reply:
```cmd
func createAPIAvailabilityMeasurement() measurement.Measurement {
	return &apiAvailabilityMeasurement{}
}
type apiAvailabilityMeasurement struct {
}
func (a *apiAvailabilityMeasurement) Dispose() {}

func (a *apiAvailabilityMeasurement) String() string {
	return "APIAvailability"
}
func (a *apiAvailabilityMeasurement) Execute(config *measurement.Config) ([]measurement.Summary, error) {
	return nil,nil
}
```

Reply:
@marseel , looks good?

Reply:
The test case runs fine , but there is another problem where in other test cases the measurement name in dummy and not registered!

Reply:
Could you point to the test in which there is this measurement dummy?